extensions/2.0/Khronos/KHR_audio_graph/README.md

```diff
@@ -0,0 +1,1279 @@
+
+
+<table>
+  <tr>
+   <td>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Notes</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>input channels
+   </td>
+   <td>integer
+   </td>
+   <td>Number of channels in input audio.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>output channels
+   </td>
+   <td>integer
+   </td>
+   <td>Number of channels in output audio.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>channel interpretation
+   </td>
+   <td>string
+   </td>
+   <td>speakers, discrete
+<p>
+Speakers use up-mix / down-mix equations. In cases where the number of channels do not match any of the basic speaker layouts, use "discrete". "Discrete" up-mix by filling channels until they run out then zero out remaining channels; down-mix by filling as many channels as possible, then dropping remaining channels.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+
+### 6.7 Audio mixer node (N inputs / 1 output)
+
+Use the Audio mixer node to combine the output from multiple audio sources. Number of channels should be the same across all inputs.
+
+[no properties for this node]
+
+
+### 6.8 Filter node (1 input / 1 output)
+
+Use the Audio Mixer node to combine the output from multiple audio sources. A filter node always has exactly one input and one output.
+
+
+<table>
+  <tr>
+   <td>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Notes</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>type
+   </td>
+   <td>string
+   </td>
+   <td>Defining the kind of filtering algorithm the node is implementing (lowpass, highpass, bandpass, lowshelf, highshelf, peaking, notch, allpass, custom)
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>frequency
+   </td>
+   <td>number
+   </td>
+   <td>Frequency in the current filtering algorithm measured in Hz.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>quality factor
+   </td>
+   <td>number
+   </td>
+   <td>The lower the Quality, the broader the bandwidth of frequencies cut or boosted. Value range: 0 to 100.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>gain
+   </td>
+   <td>number
+   </td>
+   <td>gain applied to the signal by the filter.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>bypass
+   </td>
+   <td>boolean
+   </td>
+   <td>Disables this processor while still allowing unprocessed audio signals to pass.
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+
+### 6.9 Reverb node (1 input / 1 output)
```

Comment:
Reverb implementations vary widely. Has any thought been given to consistency across implementations for this node? Web-based implementations would have to create their own reverb nodes since there is no WebAudio reverb node, yet. Would a `Convolver` node make more sense?

Reply:
Good callout, extended support for convolution reverb in addition to algorithmic reverb.

Reply:
Ok, thanks, but this doesn't address my concern about the algorithmic reverb sounding different across implementations. I don't think sound designers are going to use a reverb if it doesn't sound the same everywhere, so how do we get the algorithmic reverb to sound the same on all implementations? Can a specific reverb algorithm be required by the standard, like maybe a basic plate reverb?

Reply:
Got it. Unfortunately, there is no universal standard for reverb implementation, which can vary significantly in core business logic and its complexity. Many implementations remain unpublished. Therefore, the objective is not to restrict reverb to a particular implementation but to rather parameterize it sufficiently to accommodate most implementations. Requiring a specific reverb algorithm as a standard would, in fact, diminish the value for users. That said, one can always leverage convolution reverb with a specific impulse response.

Reply:
> Requiring a specific reverb algorithm as a standard would, in fact, diminish the value for users.

Maybe, but I think users won't create graphs with algorithmic reverbs because they won't know how they'll sound on all implementations. Instead, they'll bake the exact reverb they want directly into the audio sources. They'll do this with any audio nodes that vary significantly across implementations.

Reply:
>because they won't know how they'll sound on all implementations 

That's a valid point, and it extends beyond just different implementations to the same implementation being used across various device types, such as mobiles, desktops, and HMDs. Each device type, and indeed each variant within those types, can potentially alter the sound characteristics, making it challenging for users to predict how the reverb will sound without testing it on each device themselves. Moreover, developers often face a trade-off between computational cost and fidelity in their implementations. This balance is crucial to ensure that the reverb not only meets quality expectations but also remains computationally feasible and customizable across diverse hardware platforms.